### PR TITLE
Remove LCOW (step 1)

### PR DESCRIPTION
**Pull request by thaJeztah**
taken from https://github.com/moby/moby/pull/42170, but splitting into smaller chunks

The LCOW implementation in dockerd has been deprecated in favor of re-implementation
in containerd (in progress). Microsoft started removing the LCOW V1 code from the
build dependencies we use in Microsoft/opengcs (soon to be part of Microsoft/hcshhim),
which means that we need to start removing this code.

This first step removes the lcow graphdriver, the LCOW initialization code, and
some LCOW-related utilities.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
relates to https://github.com/moby/moby/issues/39533

**Comment by thaJeztah** [comment]
@AkihiroSuda @tonistiigi @cpuguy83 ptal

File: project/PACKAGERS.md [hunk 1]
```diff
@@ -185,12 +185,6 @@ NOTE: if you need to set more than one build tag, space separate them:
 export DOCKER_BUILDTAGS='apparmor exclude_graphdriver_aufs'
 ```
 
-### LCOW (Linux Containers On Windows)
-
-LCOW is an experimental feature on Windows, and requires the daemon to run with
-experimental features enabled. Use the `no_lcow` build tag to disable the LCOW
-feature at compile time, 
-
```
Comment (AkihiroSuda) [inline project/PACKAGERS.md:193]: (Off-topic: PACKAGERS.md seems completely outdated 😅)

Comment (thaJeztah) [reply project/PACKAGERS.md:193]: Yes, it needs to be cleaned up

File: pkg/system/path_windows.go [hunk 2]
```diff
@@ -25,3 +31,18 @@ func GetLongPathName(path string) (string, error) {
 	}
 	return windows.UTF16ToString(b), nil
 }
+
+// checkSystemDriveAndRemoveDriveLetter is the Windows implementation
+// of CheckSystemDriveAndRemoveDriveLetter
+func checkSystemDriveAndRemoveDriveLetter(path string, driver PathVerifier) (string, error) {
```
Comment (AkihiroSuda) [inline pkg/system/path_windows.go:37]: Can we just export this as CheckSystemDriveAndRemoveDriveLetter?

**Review by AkihiroSuda — APPROVED** [review]

File: pkg/system/path_windows.go (hunk 2 above)
Comment (thaJeztah) [reply pkg/system/path_windows.go:37]: We could, but the downside is we'll have to maintain GoDoc for both Windows and Linux separately, plus making sure that the signatures are in sync; having the "exported" version in a shared file, and the implementations separate makes sure we don't diverge.

**Comment by thaJeztah** [comment]
this one is failing on windows (flaky test: https://github.com/moby/moby/issues/42459)

```
=== RUN   TestNetworkDBIslands
```

**Comment by thaJeztah** [comment]
Failures on arm64:

`TestConcurrencyNoWait` tracked through https://github.com/moby/moby/issues/42458

```
=== RUN   TestConcurrencyNoWait
--- FAIL: TestConcurrencyNoWait (0.16s)
    iptables_test.go:217:  (iptables failed: iptables -t nat -A DOCKEREST -p tcp -d 192.168.1.1 --dport 1234 -j DNAT --to-destination 172.17.0.1:4321 ! -i lo: Another app is currently holding the xtables lock. Perhaps you want to use the -w option?
         (exit status 4))
```

`TestExistsRaw` created tracking issue: https://github.com/moby/moby/issues/42469

```
=== RUN   TestExistsRaw
--- FAIL: TestExistsRaw (0.10s)
    iptables_test.go:288: Failed to detect rule. i=2
```

Failure on amd64

`TestDiff` created tracking issue https://github.com/moby/moby/issues/42470

```
=== RUN   TestDiff
--- FAIL: TestDiff (1.03s)
    diff_test.go:42: assertion failed: 
        --- expected
        +++ items
        {[]container.ContainerChangeResponseItem}:
        	-: []container.ContainerChangeResponseItem{{Kind: 0x01, Path: "/foo"}, {Kind: 0x01, Path: "/foo/bar"}}
        	+: []container.ContainerChangeResponseItem(nil)
```

**Comment by thaJeztah** [comment]
This one failed again (on arm64)

```
=== RUN   TestExistsRaw
--- FAIL: TestExistsRaw (0.04s)
    iptables_test.go:288: Failed to detect rule. i=2
```

Otherwise everything is green

@cpuguy83 @tianon ptal

File: pkg/system/lcow_unsupported.go [hunk 3]
```diff
@@ -1,27 +1,14 @@
...
-	specs "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
-// InitLCOW does nothing since LCOW is a windows only feature
-func InitLCOW(_ bool) {}
-
 // LCOWSupported returns true if Linux containers on Windows are supported.
 func LCOWSupported() bool {
 	return false
 }
```
Comment (tianon) [inline pkg/system/lcow_unsupported.go:10]: Is this still used somewhere, or should this whole file likely be going away?

**Review by tianon — APPROVED** [review]
Just one minor question -- overall LGTM

The following (deleted) comment was helpful context for me while reviewing this PR:

```go
// ValidatePlatform determines if a platform structure is valid. This function
// is used for LCOW, and is a no-op on non-windows platforms.
```

(It's about halfway down the diff, but it's deleted in a lot of places and IMO this comment helps explain why so it's the best place to start. :smile:)

File: pkg/system/lcow_unsupported.go (hunk 3 above)
Comment (thaJeztah) [reply pkg/system/lcow_unsupported.go:10]: Let me double check if it's still called in this diff. Once everything is done, it should definitely be gone.

I mostly opened this PR (smaller subset of https://github.com/moby/moby/pull/42170) for my own sanity, as the LCOW code is in so many places, that "chipping it off" in smaller bits made it somewhat easier both for me, as well as for reviewers. 😓

Comment (thaJeztah) [reply pkg/system/lcow_unsupported.go:10]: Just checked, and `LCOWSupported()` is still used in various places (but now of course always `false`)

**Comment by thaJeztah** [comment]
It's ✅ green again!

File: pkg/system/lcow_unsupported.go (hunk 3 above)
Comment (tianon) [reply pkg/system/lcow_unsupported.go:10]: Sounds good! 👍

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**